subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/processors/RunSortedBySizeTestClassProcessor.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.testing.processors;
+
+import org.gradle.api.internal.tasks.testing.TestClassProcessor;
+import org.gradle.api.internal.tasks.testing.TestClassRunInfo;
+import org.gradle.api.internal.tasks.testing.TestResultProcessor;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.Map;
+
+/**
+ * This processor uses test class size information to resort test classes by
+ * size in descending order before passing them on for processing.
+ *
+ * Tests processed without size information are at the end behind known classes.
+ * If no sizes at all are known beforehand, this will not change the test order.
+ */
+public class RunSortedBySizeTestClassProcessor implements TestClassProcessor {
+	private final TestClassProcessor delegate;
+	private final Map<String, Long> sizeByTestClassName;
+
+	private final LinkedList<TestClassRunInfo> knownTestClasses = new LinkedList<TestClassRunInfo>();
+	private final LinkedHashSet<TestClassRunInfo> unknownTestClasses = new LinkedHashSet<TestClassRunInfo>();
+
+	public RunSortedBySizeTestClassProcessor(Map<String, Long> sizeByTestClassName, TestClassProcessor delegate) {
+		this.sizeByTestClassName = sizeByTestClassName;
+		this.delegate = delegate;
+	}
+
+	@Override
+	public void startProcessing(TestResultProcessor resultProcessor) {
+		delegate.startProcessing(resultProcessor);
+	}
+
+	@Override
+	public void processTestClass(TestClassRunInfo testClass) {
+		if (sizeByTestClassName.containsKey(testClass.getTestClassName())) {
+			knownTestClasses.add(testClass);
+		} else {
+			unknownTestClasses.add(testClass);
+		}
+	}
+
+	@Override
+	public void stop() {
+		// sort and process known classes first
+		Collections.sort(knownTestClasses, new Comparator<TestClassRunInfo>() {
+			@Override
+			public int compare(TestClassRunInfo t1, TestClassRunInfo t2) {
```

Comment:
Please use `list.sort(..)` and `Comparators.comparing`

Reply:
I concur and would gladly abide, but IntelliJ tells me that would be beyond the configured Language Level for the module. I don't believe I changed the level; should it be higher? Or am I missing something else?

Reply:
Oh, I thought Gradle sources are Java 8.

Reply:
Yeah, so did I, but at least according to the various automatically imported settings, it seems to be a very mixed bag. The https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md actually point out that parts still have to compatible with 1.6.

Reply:
Indeed, I thought so too; but according to the automatically imported Gradle settings, it is a mixed bag, and IntelliJ marks my usage of `list.sort(...)` as an error in the class unless I change the language level. The [contribution guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md) point out that some parts even need to be compatible with Java 1.6, so I guess that is not an option.